core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -355,7 +355,9 @@ struct CudaParallelLaunchKernelInvoker<
 
     if (!Impl::is_empty_launch(grid, block)) {
       Impl::check_shmem_request(cuda_instance, shmem);
-      Impl::configure_shmem_preference(base_t::get_kernel_func(), prefer_shmem);
+      Impl::configure_shmem_preference<DriverType, LaunchBounds,
+                                       decltype(base_t::get_kernel_func())>(
```

Comment:
I don't think you need to specify the last template argument.  I expect FTAD does the right thing.